webtraversallibrary/workflow.py

```diff
@@ -230,59 +234,68 @@ def _get_new_views(self) -> Dict[str, View]:
                 not self.config.scraping.all
                 and self.previous_policy_result
                 and tab not in self.previous_policy_result
-                and self.history[-1]
-                and self.history[-1].snapshot
+                and self.latest_view
+                and self.latest_view.snapshot
             ):
                 self.history.append(self.history[-1])
```

Comment:
This seems weird...should this be `self.history.append(self.latest_tab)` instead?